internal/cloud/gke.go

```diff
@@ -69,15 +70,29 @@ const (
 
 var _ Provider = &GKE{}
 
+// GKE is a wrapper around the GKE API.
 type GKE struct {
-	NodePools      NodePoolService
+	NodePools      NodePoolAPI
+	Reservations   ReservationProviderAPI
 	ClusterContext GKEContext
+	Recorder       record.EventRecorder
 
-	Recorder record.EventRecorder
-
-	inProgressDeletesNPName sync.Map
 	inProgressCreatesNPName sync.Map
 	inProgressCreatesJobKey sync.Map
+	inProgressDeletesNPName sync.Map
+}
+
+// NodePoolAPI is an interface for a node pool provider.
+type NodePoolAPI interface {
```

Comment:
Looks like a duplicate of `type NodePoolService interface` that is already defined.